### PR TITLE
Update dev workflow

### PR DESCRIPTION
**Pull request by astik**
- remove CSS from distribution and process as there is no CSS into this project
- replace ngmin with ngannotate (ngmin is not maintained anymore)
- add jsbeautifier to have code convention consistency (code is not yet format in this PR)
- generated files are remove from demo (they were duplicated from dist and bower_component) ; in place, in the demo, connect server serves files from dist and bower_components directly
- update and freeze version number for all dev dependencies
- bump both package.json and bower.json while releasing

### CODE REVIEW

**Comment by astik** [comment]
If you want, I can rebase this PR whenever it is needed (especially after the other PR for timeAgoSettings)

File: Gruntfile.js [hunk 1]
```diff
@@ -39,27 +41,30 @@ module.exports = function (grunt) {
...
-        dest: '<%= dirs.dist %>/<%= pkg.name %>.css'
       }
     },
 
     connect: {  // grunt-contrib-connect
+      options: {
+        port: 9999,
+        hostname: '0.0.0.0',
+        keepalive: true,
+		middleware : function(connect) {
```
Comment (yaru22) [inline Gruntfile.js]: nit: Could you indent this properly?

File: Gruntfile.js [hunk 2]
```diff
@@ -160,12 +156,18 @@ module.exports = function (grunt) {
...
-          '<%= dirs.src %>/*.js',
-          '<%= dirs.src %>/*.css',
+          '<%= dirs.src %>/*.js'
         ],
         tasks: ['test'],
       }
-    }
+    },
+
+	jsbeautifier : {
```
Comment (yaru22) [inline Gruntfile.js]: ditto: indentation

**Comment by yaru22** [comment]
@astik, I like the changes in this PR. I'll merge it in after minor indentation fixes.

File: Gruntfile.js [hunk 3]
```diff
@@ -191,10 +193,10 @@ module.exports = function (grunt) {
   // Build task.
   grunt.registerTask('build', [
     // 'test',
+    'jsbeautifier',
     'concat',
     'ngmin',
```
Comment (yaru22) [inline Gruntfile.js]: Replace ngmin with ngannotate?

**Comment by astik** [comment]
I rebase the branch in order to have no conflict for you.
I took care of indentation inconsistencies by adding Gruntfile to jsbeautifier. Now, when doing *grunt build* every source code file are formatted.
I try to follow your convention so that not many changes were made on actual code. If you want to change the convention, feel free to modify the settings files : .jsbeautifyrc

**Comment by yaru22** [comment]
Thanks for the rebase. Merging it in now.

**Comment by yaru22** [comment]
@astik: since you made a significant contribution to the codebase, I'll add you as a collaborator. Thanks for contributing again!

**Comment by astik** [comment]
Thank you for adding me.
I'll add some more PR. I'll let you check them, if you agree with them, add a notification (like "ok to merge" or something like that), then I'll take care of the if-needed rebase and merge.

**Pull request merged**